### ISSUE TITLE
model.tune(): Fitness curve smoothing method and dataset used (validation vs test)?

### ISSUE THREAD

**Issue by Aditya-365** [ultralytics/ultralytics#24280]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

## Description
While analyzing the results of `model.tune()`, I noticed that the fitness curve appears to be smoothed.

## Questions

1. **Smoothing method**
   - How is the fitness curve smoothed internally?
   - Is it using a moving average, median filter, exponential smoothing, or another method?

2. **Dataset used for fitness**
   - Is the fitness score computed on the **validation set** or the **test set**?
   - I want to confirm that tuning does not use test data.

3. This  is my current fitness graph: 
    - Is this plateau behavior expected for the default genetic search strategy around the 40-iteration mark, or should I be seeing a more consistent upward trendline?

<img width="460" height="278" alt="Image" src="https://github.com/user-attachments/assets/6dfdfe06-2b59-4d29-a94d-70597b57f28e" />

## What I ran
~~~python
from ultralytics import YOLO

model = YOLO("yolo12n.pt")
model.tune(data="data.yaml", epochs=50, iterations=40, batch=8, img_size=640, use_ray=False)
~~~

## Environment
OS: Linux
Python: 3.11.15
Ultralytics: 8.3.63
Torch: 2.7.0+cu128
GPU: NVIDIA RTX 5090

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Aditya-365, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
For your `model.tune()` question specifically, it would help if you could share:
- The exact command or script you used
- The full console output/logs from tuning
- Any saved tuning results files/plots from the run
- Whether this behavior is reproducible across multiple runs or datasets
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Yes, that’s expected: the dashed fitness line in [`plot_tune_results()`](https://docs.ultralytics.com/reference/utils/plotting/) is smoothed with `scipy.ndimage.gaussian_filter1d(..., sigma=3)`, not a moving average/EMA, and the fitness used by [`Tuner.__call__`](https://docs.ultralytics.com/reference/engine/tuner/) comes from the training run’s `train_metrics`, which are based on the dataset YAML’s `val` split rather than `test` unless you explicitly point `val` at test data; a plateau around ~40 iterations is normal for the default mutation-based search, so improvements are usually noisy and non-monotonic, and for new runs I’d recommend switching from `yolo12n.pt` to `yolo26n.pt`.
